src/config/syscheck-config.c

```diff
@@ -261,6 +230,45 @@ void dump_syscheck_entry(syscheck_config *syscheck, char *entry, int vals, int r
             os_strdup(tag, syscheck->tag[pl]);
         }
     }
+#ifdef WIN32
+    else {
+        if (syscheck->registry == NULL) {
+            os_calloc(2, sizeof(registry), syscheck->registry);
+            syscheck->registry[pl + 1].entry = NULL;
+            syscheck->registry[pl].tag = NULL;
+            syscheck->registry[pl + 1].tag = NULL;
+            syscheck->registry[pl].arch = vals;
+            os_strdup(entry, syscheck->registry[pl].entry);
+        } else if (overwrite < 0) {
+            bool duplicated_registry = false;
+            while (syscheck->registry[pl].entry != NULL) {
+                /* Duplicated entry */
+                if (strcmp(syscheck->registry[pl].entry, entry) == 0 && vals == syscheck->registry[pl].arch) {
+                    duplicated_registry = true;
+                    mdebug2("Duplicated registration entry: %s", syscheck->registry[pl].entry);
+                    break;
+                }
+                pl++;
+            }
+            if (!duplicated_registry){
+                os_realloc(syscheck->registry, (pl + 2) * sizeof(registry),
+                        syscheck->registry);
+                syscheck->registry[pl + 1].entry = NULL;
+                syscheck->registry[pl].tag = NULL;
+                syscheck->registry[pl + 1].tag = NULL;
+                syscheck->registry[pl].arch = vals;
+                os_strdup(entry, syscheck->registry[pl].entry);
+            }
+        } else {
+            os_free(syscheck->registry[pl].tag);
+            syscheck->registry[pl].arch = vals;
+        }
+
+        if (tag) {
+            os_strdup(tag, syscheck->registry[pl].tag);
```

Comment:
At this point with a duplicate record, there is a memory leak as the tag from the previous record (line 263) is never released.